src/ray/raylet/worker_pool_test.cc

```diff
@@ -1348,6 +1406,119 @@ TEST_F(WorkerPoolTest, PopWorkerWithRuntimeEnv) {
   ASSERT_EQ(worker_pool_->GetProcessSize(), 3);
 }
 
+TEST_F(WorkerPoolTest, RuntimeEnvUriReference) {
+  valid_uris.clear();
```

Comment:
Can you add an unit test for job start / removed?

Reply:
Add test here https://github.com/ray-project/ray/pull/20789/commits/884188c1b27d58e412b2f172fd5d7cb7c3e704eb#diff-bf9523ff768067f9793ae89d37ff19c53c94ff450e39fd20eaf23a57dd38ccbcR1412